code_challenges.rb

```diff
@@ -1,153 +1,48 @@
 # ASSESSMENT 4: Ruby Coding Practical Questions
 
 # --------------------1) Create a method that takes in an array and returns an array with only odd numbers sorted from least to greatest.
-puts "1:\n"
 
-full_arr1 = [4, 9, 0, '7', 8, true, 'hey', 7, 199, -9, false, 'hola']
+fullArr1 = [4, 9, 0, '7', 8, true, 'hey', 7, 199, -9, false, 'hola']
 # Expected output: [-9, 7, 9, 199]
-full_arr2 = ['hello', 7, 23, -823, false, 78, nil, '67', 6, 'Number']
+fullArr2 = ['hello', 7, 23, -823, false, 78, nil, '67', 6, 'Number']
 # Expected output: [-823, 7, 23]
 
-ef median(array)
-  return nil if array.empty?
-  sorted = array.sort
-  len = sorted.length
-  (sorted[(len - 1) / 2] + sorted[len / 2]) / 2.0
-end
-end
 
-puts median(full_arr1)
 
-# This one was super hard and again, I did not do super well with it. I had some pieces working but the more I added and changed the more broken it got and the more I forgot how to even get back to what I had.
 
-# --------------------2) Create a class called Bike that is initialized with a model, wheels, and a frame size. The default number of wheels is 2. Create a get_info method that returns a sentence with all the data from the bike object.
-puts "2:\n"
-
-class Bike 
-    def initialize (model, wheels, frame_size)
-        @model = model
-        @wheels = wheels 
-        @frame_size = frame_size
-    end
-
-    def get_model
-        @model 
-    end
 
-    def set_model=(model)
-        @model = model
-    end
-
-    def get_wheels
-        @model 
-    end
-
-    def set_wheels=(wheels)
```

Comment:
This part is really close. Couple things...
- Since the default number of wheels is 2 you can skip the setter method and just assign the value in on line 29 to be 2.
- According to the prompt, we only really need one "getter" method that returns a sentence with the bike info. So while the methods for get model, and get wheels are correct, they just aren't necessary. 
- The initializer is sort of a fancy setter method. Since the model and frame size are getting passed in on creation of a new object, the values can be taken from the argument of .new
```
my_bike = Bike.new("Trek bike", "168cm frame")
```
Then the instance variable of @model and @frame_size will have the appropriate values.